pkg/config/config.go

```diff
@@ -0,0 +1,625 @@
+
+	return uniqueSecretsIDs, cobra.ShellCompDirectiveNoFileComp
+}
+
+func (config *Config) LoadEnvironmentID(environmentName string) (string, error) {
+	aa, err := config.Cloud.Environments(config.Ctx, config.ProjectID, cloud.EnvironmentsParams{
+		Name: &environmentName,
+		Last: Ptr(uint(1)),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	if len(aa.Items) == 0 {
+		return "", fmt.Errorf("could not find environment %q", environmentName)
+
+	}
+
+	return aa.Items[0].ID, nil
+}
+
+func (config *Config) LoadCoreInstanceID(key string, environmentID string) (string, error) {
+	params := cloud.CoreInstancesParams{
+		Name: &key,
+		Last: Ptr(uint(2)),
+	}
+
+	if environmentID != "" {
+		params.EnvironmentID = &environmentID
+	}
+
+	aa, err := config.Cloud.CoreInstances(config.Ctx, config.ProjectID, params)
+	if err != nil {
+		return "", err
+	}
+
+	if len(aa.Items) != 1 && !ValidUUID(key) {
+		if len(aa.Items) != 0 {
+			return "", fmt.Errorf("ambiguous core instance name %q, use ID instead", key)
+		}
+
+		return "", fmt.Errorf("could not find core instance %q", key)
+	}
+
+	if len(aa.Items) == 1 {
+		return aa.Items[0].ID, nil
+	}
+
+	return key, nil
+}
+
+func (config *Config) LoadPipelineID(pipelineKey string) (string, error) {
+	pp, err := config.Cloud.ProjectPipelines(config.Ctx, config.ProjectID, cloud.PipelinesParams{
+		Name: &pipelineKey,
+		Last: Ptr(uint(2)),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	if len(pp.Items) != 1 && !ValidUUID(pipelineKey) {
+		if len(pp.Items) != 0 {
+			return "", fmt.Errorf("ambiguous pipeline name %q, use ID instead", pipelineKey)
+		}
+
+		return "", fmt.Errorf("could not find pipeline %q", pipelineKey)
+	}
+
+	if len(pp.Items) == 1 {
+		return pp.Items[0].ID, nil
+	}
+
+	return pipelineKey, nil
+}
+
+
+func (config *Config) FetchAllPipelines() ([]cloud.Pipeline, error) {
+	aa, err := config.Cloud.CoreInstances(config.Ctx, config.ProjectID, cloud.CoreInstancesParams{})
+	if err != nil {
+		return nil, fmt.Errorf("could not prefetch core-instances: %w", err)
+	}
+
+	if len(aa.Items) == 0 {
+		return nil, nil
+	}
+
+	var pipelines []cloud.Pipeline
+	var mu sync.Mutex
+
+	g, gctx := errgroup.WithContext(config.Ctx)
+	for _, a := range aa.Items {
+		a := a
+		g.Go(func() error {
+			got, err := config.Cloud.Pipelines(gctx, a.ID, cloud.PipelinesParams{})
+			if err != nil {
+				return err
+			}
+
+			mu.Lock()
+			pipelines = append(pipelines, got.Items...)
+			mu.Unlock()
+
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+
+	var uniquePipelines []cloud.Pipeline
+	pipelineIDs := map[string]struct{}{}
+	for _, pip := range pipelines {
+		if _, ok := pipelineIDs[pip.ID]; !ok {
+			uniquePipelines = append(uniquePipelines, pip)
+			pipelineIDs[pip.ID] = struct{}{}
+		}
+	}
+
+	return uniquePipelines, nil
+}
+
+func (config *Config) CompletePipelinePlugins(pipelineKey string, cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

Comment:
This should be part of the completer package too.

Reply:
Didn't get to it yet. It is next in the line